sdk/src/main/java/io/opentdf/platform/sdk/Autoconfigure.java

```diff
@@ -0,0 +1,749 @@
+            Pattern pattern = Pattern.compile("^(https?://[\\w./-]+/attr/\\S*)/value/\\S*$");
+            Matcher matcher = pattern.matcher(url);
+            if (!matcher.find()) {
+                throw new AutoConfigureException("invalid type");
+            }
+            return new AttributeNameFQN(matcher.group(1));
+        }
+
+        public String value() throws AutoConfigureException {
+            Pattern pattern = Pattern.compile("^https?://[\\w./-]+/attr/\\S*/value/(\\S*)$");
+            Matcher matcher = pattern.matcher(url);
+            if (!matcher.find()) {
+                throw new AutoConfigureException("invalid type");
+            }
+            try {
+                return URLDecoder.decode(matcher.group(1), StandardCharsets.UTF_8.name());
+            } catch (Exception e) {
+                throw new AutoConfigureException("invalid type");
+            }
+        }
+
+        public String name() throws AutoConfigureException {
+            Pattern pattern = Pattern.compile("^https?://[\\w./-]+/attr/(\\S*)/value/\\S*$");
+            Matcher matcher = pattern.matcher(url);
+            if (!matcher.find()) {
+                throw new AutoConfigureException("invalid attributeInstance");
+            }
+            try {
+                return URLDecoder.decode(matcher.group(1), StandardCharsets.UTF_8.name());
+            } catch (Exception e) {
+                throw new AutoConfigureException("invalid attributeInstance");
+            }
+        }
+    }
+
+    public static class KeyAccessGrant {
+        public Attribute attr;
+        public List<String> kases;
+
+        public KeyAccessGrant(Attribute attr, List<String> kases) {
+            this.attr = attr;
+            this.kases = kases;
+        }
+    }
+
+    // Structure capable of generating a split plan from a given set of data tags.
+    public static class Granter {
+        private final List<AttributeValueFQN> policy;
+        private Map<String, KeyAccessGrant> grants = new HashMap<>();
+
+        public Granter(List<AttributeValueFQN> policy) {
+            this.policy = policy;
+        }
+
+        public Granter(List<AttributeValueFQN> policy, Map<String, KeyAccessGrant> grants) {
+            this.policy = policy;
+            this.grants = grants;
+        }
+
+        public  Map<String, KeyAccessGrant> getGrants() {
+            return grants;
+        }
+
+        public List<AttributeValueFQN> getPolicy() {
+            return policy;
+        }
+
+        public void addGrant(AttributeValueFQN fqn, String kas, Attribute attr) {
+            grants.computeIfAbsent(fqn.key, k -> new KeyAccessGrant(attr, new ArrayList<>()))
+                .kases.add(kas);
+        }
+
+        public void addAllGrants(AttributeValueFQN fqn, List<KeyAccessServer> gs, Attribute attr) {
+            if (gs.isEmpty()) {
+                grants.putIfAbsent(fqn.key, new KeyAccessGrant(attr, new ArrayList<>()));
```

Comment:
if you implement hashCode and equals on `AttributeValueFQN` on just `key`, you could use the fqn as the key instead of the string